src/test/java/reactor/netty/instrumentation/zipkin/HttpServerTracingTest.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (c) 2011-2019 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.netty.instrumentation.zipkin;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+import brave.Tracing;
+import brave.internal.Platform;
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import reactor.core.publisher.Mono;
+import reactor.netty.DisposableServer;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.server.HttpServer;
+import reactor.netty.http.server.HttpServerRequest;
+import reactor.netty.http.server.HttpServerResponse;
+import reactor.netty.http.server.HttpServerRoutes;
+import reactor.test.StepVerifier;
+import zipkin2.Annotation;
+import zipkin2.Span;
+import zipkin2.Span.Kind;
+
+public class HttpServerTracingTest {
```

Comment:
it is easy to miss edge cases, eventhough ad-hoc tests are good for understanding things locally. It would be nice to add the tests we use for all frameworks, to help understand what works vs doesn't.

ex https://github.com/openzipkin/brave/tree/master/instrumentation/vertx-web/src/test/java/brave/vertx/web

which implies a test dependency on
```
    <dependency>
      <groupId>io.zipkin.brave</groupId>
      <artifactId>brave-instrumentation-http-tests</artifactId>
      <scope>test</scope>
</dependency>
```

Reply:
Thanks, `http-tests` found some mistakes